### PR TITLE
Adds processes check for windows to collect queue length and count

### PR DESCRIPTION
**Pull request by nmuesch**
### What does this PR do?

Adds the proc.queue.length and proc.count metrics to Agent 6 on Windows. 

### Motivation

Makes this more aligned with the metrics submitted by Agent 5. We also use these metrics to help determine a host is running on Windows for some extra UI components. 

### Additional Notes

### CODE REVIEW

File: cmd/agent/dist/conf.d/processes.d/conf.yaml.default [hunk 1]
```diff
@@ -0,0 +1,2 @@
+instances:
```
Comment (masci) [inline cmd/agent/dist/conf.d/processes.d/conf.yaml.default]: we already have a `process` check, can we call this `winproc` or something like this?

**Review by masci — CHANGES_REQUESTED** [review]

File: omnibus/config/software/datadog-agent-finalize.rb [hunk 2]
```diff
@@ -66,12 +66,18 @@
             # remove unused configs
             delete "/etc/datadog-agent/conf.d/apm.yaml.default"
 
+            # remove windows specific configs
+            delete "#{install_dir}/etc/conf.d/winproc.d"
```
Comment (olivielpeau) [inline omnibus/config/software/datadog-agent-finalize.rb]: we should remove `/etc/datadog-agent/conf.d/winproc.d` here, at this point we've already moved all of `#{install_dir}/etc/datadog-agent/conf.d` to `/etc/datadog-agent`

**Review by olivielpeau — COMMENTED** [review]
needs one change, other than that LGTM

**Review by masci — DISMISSED** [review]

**Comment by olivielpeau** [comment]
CI is failing on linter step (for good reasons), fixing it

**Review by masci — APPROVED** [review]
LGTM

**Comment by olivielpeau** [comment]
Looks good, merging this

**Pull request merged**